Xamarin.Forms.Platform.Android/FormsApplicationActivity.cs

```diff
@@ -84,27 +80,18 @@ public void SetPage(Page page)
 		protected void LoadApplication(Application application)
 		{
 			if (application == null)
-				throw new ArgumentNullException("application");
+				throw new ArgumentNullException(nameof(application));
 
-			(application as IApplicationController)?.SetAppIndexingProvider(new AndroidAppIndexProvider(this));
+			(application as IApplicationController).SetAppIndexingProvider(new AndroidAppIndexProvider(this));
 
 			_application = application;
 			Xamarin.Forms.Application.SetCurrentApplication(application);
 
 			SetSoftInputMode();
 
+			application.PropertyChanged -= AppOnPropertyChanged;
 			application.PropertyChanged += AppOnPropertyChanged;
 
-			if (application?.MainPage != null)
```

Comment:
see ^^^

Reply:
You nuked the disposing. What is there that is done to replace this logic?

Reply:
The dispose is already done later in the Platform class.
This code was added later by PR #1200 and #1212 but the fix was wrong..
If you not dispose the page in the case it's the same page the issue that PR tried to fix doesn't exist anymore.